conf/helmfile.d/0600.prometheus-operator.yaml

```diff
@@ -1380,188 +191,90 @@ releases:
-          ## possibility to break upgrades of Prometheus. It is advised to review Prometheus release notes to ensure that no incompatible alert relabel
-          ## configs are going to break Prometheus after the upgrade.
-          ##
-          additionalAlertRelabelConfigs: []
-          # - separator: ;
-          #   regex: prometheus_replica
-          #   replacement: $1
-          #   action: labeldrop
-
-          ## SecurityContext holds pod-level security attributes and common container settings.
-          ## This defaults to non root user with uid 1000 and gid 2000.
-          ## https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md
-          ##
-          securityContext:
-            runAsNonRoot: true
-            runAsUser: 1000
-            fsGroup: 2000
-
-          ## 	Priority class assigned to the Pods
-          ##
-          priorityClassName: ""
-
-          ## Thanos configuration allows configuring various aspects of a Prometheus server in a Thanos environment.
-          ## This section is experimental, it may change significantly without deprecation notice in any release.
-          ## This is experimental and may change significantly without backward compatibility in any release.
-          ## ref: https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#thanosspec
-          ##
-          thanos: {}
-
-          ## Containers allows injecting additional containers. This is meant to allow adding an authentication proxy to a Prometheus pod.
-          ##  if using proxy extraContainer  update targetPort with proxy container port
-          containers: []
-
-          ## Enable additional scrape configs that are managed externally to this chart. Note that the prometheus
-          ## will fail to provision if the correct secret does not exist.
-          ##
-          additionalScrapeConfigsExternal: false
-
-        additionalServiceMonitors: []
-        ## Name of the ServiceMonitor to create
-        ##
-        # - name: ""
-
-          ## Additional labels to set used for the ServiceMonitorSelector. Together with standard labels from
-          ## the chart
-          ##
-          # additionalLabels: {}
-
-          ## Service label for use in assembling a job name of the form <label value>-<port>
-          ## If no label is specified, the service name is used.
-          ##
-          # jobLabel: ""
-
-          ## Label selector for services to which this ServiceMonitor applies
-          ##
-          # selector: {}
-
-          ## Namespaces from which services are selected
-          ##
-          # namespaceSelector:
-            ## Match any namespace
-            ##
-            # any: false
-
-            ## Explicit list of namespace names to select
-            ##
-            # matchNames: []
+      ## Component scraping the kubelet and kubelet-hosted cAdvisor
```

Comment:
Can you explain what the `metricRelabelings` are for?

Reply:
We're using it to filter out some of the default rules that we do not use and take up a lot of memory (#314). Prometheus can really eat up A LOT of memory, so we were trying to reduce the memory footprint as much as possible.

We also use metric relabelings to add labels to the redis-script output, making it super easy to combine metrics:

```
record: consumers_per_gpu
              expr: |-
                kube_deployment_status_replicas_available{deployment=~".*-consumer"}
                / on() group_left
                max(
                  kube_deployment_status_replicas_available{deployment="tf-serving"}
                  or vector(1)
                )
```

Reply:
Ok, this makes a lot of sense. Thanks for the explanation.